### PR TITLE
[JBIDE-27073] Add user agent string to request headers

### PR DESCRIPTION
**Pull request by robstryker**
Signed-off-by: Rob Stryker <rob@oxbeef.net>

### CODE REVIEW

**Comment by adietish** [comment]
@robstryker can you please specify what issue this is fixing?

**Comment by robstryker** [comment]
https://issues.redhat.com/browse/JBIDE-27073

**Comment by adietish** [comment]
Didn't test it, code looks good to me. The only thing I'd consider is splitting up ```InternalURLTransport#transfer``` into smaller chunks that one can digest more easily.

File: foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/ecf/internal/InternalURLTransport.java [hunk 1]
```diff
@@ -366,8 +369,9 @@ private IStatus transfer(final String name,final IRetrieveFileTransferContainerA
 					String userCredentials = ncb.getName()+ ":" + pcb.getPassword();
 					String basicAuth = "Basic " + new String(new Base64().encode(userCredentials.getBytes()));
 					headers.put("Authorization", basicAuth);
-					map.put(IRetrieveFileTransferOptions.REQUEST_HEADERS, headers);
 				}
+				headers.put("User-Agent", generateUserAgent()); //$NON-NLS-1$
+				map.put(IRetrieveFileTransferOptions.REQUEST_HEADERS, headers);
 			} catch(IOException | UnsupportedCallbackException ce) {
 			}
```
Comment (adietish) [inline foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/ecf/internal/InternalURLTransport.java]: Shouldn't we at least log the exception?

File: foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/ecf/internal/InternalURLTransport.java [hunk 2]
```diff
@@ -366,8 +369,9 @@ private IStatus transfer(final String name,final IRetrieveFileTransferContainerA
...
 		}
```
Comment (adietish) [inline foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/ecf/internal/InternalURLTransport.java]: This method is a bit long, I'd maybe split it up in smaller chunks that one can digest more easily.

**Comment by adietish** [comment]
I'd also add a reference to the JBIDE issue in the commit msg so that this commit can be more easily tracked back to the issue it is solving.

File: foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/ecf/internal/InternalURLTransport.java [hunk 3]
```diff
@@ -400,6 +405,28 @@ private IStatus transfer(final String name,final IRetrieveFileTransferContainerA
...
+	protected String generateUserAgent() {
+		String prodName = null;
+		String version = null;
+		IPropertiesProvider productInfo = PropertiesHelper.getPropertiesProvider();
+		if(productInfo instanceof VersionPropertiesProvider ) {
+			VersionPropertiesProvider prov = (VersionPropertiesProvider)productInfo;
+			prodName = prov.getContext();
+			version = prov.getCurrentVersion();
+		} else {
+			prodName = "unknown";
```
[the remark this answers is not in the document]
Comment (robstryker) [reply foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/ecf/internal/InternalURLTransport.java]: Fixed

**Comment by jeffmaury** [comment]
testPR

**Pull request merged**